src/query/api/v1/handler/influxdb/rewrite.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package influxdb
+
+import (
+	"regexp"
+)
+
+type regexpRewriter struct {
+	okStart, okRest [256]bool
+	replacement     byte
+}
+
+func newRegexpRewriter(startRe, restRe string) *regexpRewriter {
+	createArray := func(okRe string) (ret [256]bool) {
+		re := regexp.MustCompile(okRe)
+		// Check for only 7 bit non-control ASCII characters
+		for i := 32; i < 128; i++ {
+			if re.Match([]byte{byte(i)}) {
+				ret[i] = true
+			}
+		}
+		return
+	}
```

Comment:
nit: can we drop regex from this completely? i.e. generate this bool array from a list of valid characters directly? The fewer moving parts the better :+1: